### ISSUE TITLE
Z-stack end position

### ISSUE THREAD

**Issue by JinlongL** [TheDeanLab/ASLM#572]
I tried to adjust the wrong setting, and it triggered this error:

```

Traceback (most recent call last):
  File "C:\Users\Dean-Lab\miniconda3\envs\ASLM\lib\tkinter\__init__.py", line 1892, in __call__
    return self.func(*args)
  File "c:\users\dean-lab\desktop\aslm\src\aslm\controller\sub_controllers\tiling_wizard_controller.py", line 177, in <lambda>
    "write", lambda *args: self.update_fov()
  File "c:\users\dean-lab\desktop\aslm\src\aslm\controller\sub_controllers\tiling_wizard_controller.py", line 539, in update_fov
    z = float(self.stack_acq_widgets["end_position"].get()) - float(
ValueError: could not convert string to float: ''
Exception in Tkinter callback
Traceback (most recent call last):
  File "c:\users\dean-lab\desktop\aslm\src\aslm\controller\sub_controllers\channels_tab_controller.py", line 459, in update_z_steps
    start_position = float(self.stack_acq_vals["start_position"].get())
  File "C:\Users\Dean-Lab\miniconda3\envs\ASLM\lib\tkinter\__init__.py", line 568, in get
    return self._tk.getdouble(self._tk.globalgetvar(self._name))
_tkinter.TclError: expected floating-point number but got "-"

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Dean-Lab\miniconda3\envs\ASLM\lib\tkinter\__init__.py", line 1892, in __call__
    return self.func(*args)
  File "c:\users\dean-lab\desktop\aslm\src\aslm\controller\sub_controllers\tiling_wizard_controller.py", line 174, in <lambda>
    "write", lambda *args: self.update_fov()
  File "c:\users\dean-lab\desktop\aslm\src\aslm\controller\sub_controllers\tiling_wizard_controller.py", line 539, in update_fov
    z = float(self.stack_acq_widgets["end_position"].get()) - float(
ValueError: could not convert string to float: ''
Exception in Tkinter callback
Traceback (most recent call last):
  File "c:\users\dean-lab\desktop\aslm\src\aslm\controller\sub_controllers\channels_tab_controller.py", line 459, in update_z_steps
    start_position = float(self.stack_acq_vals["start_position"].get())
  File "C:\Users\Dean-Lab\miniconda3\envs\ASLM\lib\tkinter\__init__.py", line 568, in get
    return self._tk.getdouble(self._tk.globalgetvar(self._name))
_tkinter.TclError: expected floating-point number but got "-"

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Dean-Lab\miniconda3\envs\ASLM\lib\tkinter\__init__.py", line 1892, in __call__
    return self.func(*args)
  File "c:\users\dean-lab\desktop\aslm\src\aslm\controller\sub_controllers\tiling_wizard_controller.py", line 177, in <lambda>
    "write", lambda *args: self.update_fov()
  File "c:\users\dean-lab\desktop\aslm\src\aslm\controller\sub_controllers\tiling_wizard_controller.py", line 539, in update_fov
    z = float(self.stack_acq_widgets["end_position"].get()) - float(
ValueError: could not convert string to float: ''
Exception in Tkinter callback
Traceback (most recent call last):
  File "c:\users\dean-lab\desktop\aslm\src\aslm\controller\sub_controllers\channels_tab_controller.py", line 459, in update_z_steps
    start_position = float(self.stack_acq_vals["start_position"].get())
  File "C:\Users\Dean-Lab\miniconda3\envs\ASLM\lib\tkinter\__init__.py", line 568, in get
    return self._tk.getdouble(self._tk.globalgetvar(self._name))
_tkinter.TclError: expected floating-point number but got "-"

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Dean-Lab\miniconda3\envs\ASLM\lib\tkinter\__init__.py", line 1892, in __call__
    return self.func(*args)
  File "c:\users\dean-lab\desktop\aslm\src\aslm\controller\sub_controllers\tiling_wizard_controller.py", line 174, in <lambda>
    "write", lambda *args: self.update_fov()
  File "c:\users\dean-lab\desktop\aslm\src\aslm\controller\sub_controllers\tiling_wizard_controller.py", line 539, in update_fov
    z = float(self.stack_acq_widgets["end_position"].get()) - float(
ValueError: could not convert string to float: ''
Exception in Tkinter callback
Traceback (most recent call last):
  File "c:\users\dean-lab\desktop\aslm\src\aslm\controller\sub_controllers\channels_tab_controller.py", line 459, in update_z_steps
    start_position = float(self.stack_acq_vals["start_position"].get())
  File "C:\Users\Dean-Lab\miniconda3\envs\ASLM\lib\tkinter\__init__.py", line 568, in get
    return self._tk.getdouble(self._tk.globalgetvar(self._name))
_tkinter.TclError: expected floating-point number but got "-"

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Dean-Lab\miniconda3\envs\ASLM\lib\tkinter\__init__.py", line 1892, in __call__
    return self.func(*args)
  File "c:\users\dean-lab\desktop\aslm\src\aslm\controller\sub_controllers\tiling_wizard_controller.py", line 177, in <lambda>
    "write", lambda *args: self.update_fov()
  File "c:\users\dean-lab\desktop\aslm\src\aslm\controller\sub_controllers\tiling_wizard_controller.py", line 539, in update_fov
    z = float(self.stack_acq_widgets["end_position"].get()) - float(
ValueError: could not convert string to float: ''
Exception in Tkinter callback
Traceback (most recent call last):
  File "c:\users\dean-lab\desktop\aslm\src\aslm\controller\sub_controllers\channels_tab_controller.py", line 460, in update_z_steps
    end_position = float(self.stack_acq_vals["end_position"].get())
  File "C:\Users\Dean-Lab\miniconda3\envs\ASLM\lib\tkinter\__init__.py", line 568, in get
    return self._tk.getdouble(self._tk.globalgetvar(self._name))
_tkinter.TclError: expected floating-point number but got ""

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Dean-Lab\miniconda3\envs\ASLM\lib\tkinter\__init__.py", line 1892, in __call__
    return self.func(*args)
  File "c:\users\dean-lab\desktop\aslm\src\aslm\controller\sub_controllers\tiling_wizard_controller.py", line 174, in <lambda>
    "write", lambda *args: self.update_fov()
  File "c:\users\dean-lab\desktop\aslm\src\aslm\controller\sub_controllers\tiling_wizard_controller.py", line 539, in update_fov
    z = float(self.stack_acq_widgets["end_position"].get()) - float(
ValueError: could not convert string to float: ''
Exception in Tkinter callback
Traceback (most recent call last):
  File "c:\users\dean-lab\desktop\aslm\src\aslm\controller\sub_controllers\channels_tab_controller.py", line 460, in update_z_steps
    end_position = float(self.stack_acq_vals["end_position"].get())
  File "C:\Users\Dean-Lab\miniconda3\envs\ASLM\lib\tkinter\__init__.py", line 568, in get
    return self._tk.getdouble(self._tk.globalgetvar(self._name))
_tkinter.TclError: expected floating-point number but got ""

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Dean-Lab\miniconda3\envs\ASLM\lib\tkinter\__init__.py", line 1892, in __call__
    return self.func(*args)
  File "c:\users\dean-lab\desktop\aslm\src\aslm\controller\sub_controllers\tiling_wizard_controller.py", line 177, in <lambda>
    "write", lambda *args: self.update_fov()
  File "c:\users\dean-lab\desktop\aslm\src\aslm\controller\sub_controllers\tiling_wizard_controller.py", line 539, in update_fov
    z = float(self.stack_acq_widgets["end_position"].get()) - float(
ValueError: could not convert string to float: ''
```

**Comment by JinlongL** [comment]
More specifically, if I try to change the values in these widgets, I get a string conversion error. May not be a validated widget. And, if we aren't actually changing the values here, but only reporting them, then perhaps they should be read only.

![image](https://github.com/TheDeanLab/ASLM/assets/96149064/c9ab6327-5072-4e5f-98b9-53d171977a6d)

**Comment by AdvancedImagingUTSW** [comment]
Currently not able to replicate this problem on my personal machine. Will evaluate on a microscope next time I am there.

**Comment by AdvancedImagingUTSW** [comment]
Addressed in #816
